lib/controller/controller.go

```diff
@@ -1055,12 +1059,35 @@ func (ctrl *ProvisionController) provisionClaimOperation(claim *v1.PersistentVol
 		return err
 	}
 
+	var selectedNode *v1.Node
+	var allowedTopologies []v1.TopologySelectorTerm
+	if ctrl.kubeVersion.AtLeast(utilversion.MustParseSemantic("v1.11.0")) {
+		// Get SelectedNode
+		if nodeName, ok := claim.Annotations[annSelectedNode]; ok {
+			selectedNode, err = ctrl.client.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{}) // TODO (verult) cache Nodes
+			if err != nil {
+				strerr := fmt.Sprintf("Failed to get target node: %v", err)
+				glog.Infof("unexpected error getting target node %q for claim %q: %v", nodeName, claimToClaimKey(claim), err)
+				ctrl.eventRecorder.Event(claim, v1.EventTypeWarning, "ProvisioningFailed", strerr)
+				return err
```

Comment:
can we use the same error message for the Event + return?

Reply:
The behavior in this block matches the behavior in the internal PVController

Reply:
agreed let's make error message same for event + return + glog, we do the same weird thing here https://github.com/verult/external-storage/blob/5dee038e0ee3133a8e40fbe39d915cef0c722af2/lib/controller/controller.go#L1102

Reply:
ACK